pre_commit/commands/run.py

```diff
@@ -224,7 +224,14 @@ def _run_hooks(config, hooks, args, environ):
                 '`pre-commit install`.',
             )
         output.write_line('All changes made by hooks:')
-        subprocess.call(('git', '--no-pager', 'diff', '--no-ext-diff'))
+        if args.color:
```

Comment:
the color option should always be present so you can just pass that through

Reply:
That is what I had put in initially, but the test `test_show_diff_on_failure` fails, cause over there `args.color` is `False`.

So I had added another entry with `color: 'auto'` in the test case so that both the paths are tested.

Now, 
1. I'm passing color unconditionally, as you suggested **and**
2. I have added `color: 'auto'` both **existing** test parameter, so that the test does not fail.

Reply:
ah ok yeah you're right -- I forgot that `pre_commit.color` translates the `('always', 'auto', 'never')` into booleans

Let me take another look at this then 👍

Reply:
I don't understand what this means. Can you point me to the relevant code ?

Also, if `args.color` is boolean, then how can be pass the "correct" value to git command ?

I think my latest changes are correct i.e. unconditionally pass the `color` option to git, and fix the test.

Reply:
ah yes

So the color option is set up [here](https://github.com/pre-commit/pre-commit/blob/fff3ad518cb9a9bf169346aa4727fb70a67a4e30/pre_commit/main.py#L40) -- this is the argument parsing bit for pre-commit.  The `type=color.use_color` tells argparse to use that callback to determine the actual value of that argument

The `use_color` function is defined [here](https://github.com/pre-commit/pre-commit/blob/fff3ad518cb9a9bf169346aa4727fb70a67a4e30/pre_commit/color.py#L42-L54) where it translates the options into booleans

You're right you do need to always pass it and not use the code sample I put below

One quick way to translate that would be to do something like this:

```python
'--color={}'.format({True: 'always', False: 'never'}[args.color])
```

This'll map the boolean values to the string options and format that into the `--color` call.

Adding that all together I think you end up with something like this

```python
subprocess.call((
    'git', '--no-pager', 'diff', '--no-ext-diff',
    '--color={}'.format({True: 'always', False: 'never'}[args.color]),
))
```

Reply:
Done.